### ISSUE TITLE
ENH: Series between inclusive end point bifurcation

### ISSUE THREAD

**Issue by colin1alexander** [pydata/pandas#12398]
Allow ability to distinguish inclusiveness of left and right boundaries.  I've written the code and tests, but need a GH tracking number prior to submitting a PR.

```
    def between(self, left, right, inclusive=True):
        """
        Return boolean Series equivalent to left <= series <= right. NA values
        will be treated as False

        Parameters
        ----------
        left : scalar
            Left boundary
        right : scalar
            Right boundary
        inclusive : boolean for both left and right or iterable pair.
            Whether or not to include the left and right boundary.
            If a tuple pair, the boundaries can be set separately
            (e.g. (False, True) for left < series <= right).
```

**Comment by jreback** [comment]
this should not be a separate method, *maybe* as an argument to ``.loc``.

**Comment by colin1alexander** [comment]
Something like this:

```
    def between(self, left, right, inclusive=True):
        """
        Return boolean Series equivalent to left <= series <= right. NA values
        will be treated as False

        Parameters
        ----------
        left : scalar
            Left boundary
        right : scalar
            Right boundary
        inclusive : boolean for both left and right or iterable pair.
            Whether or not to include the left and right boundary.
            If a tuple pair, the boundaries can be set separately
            (e.g. (False, True) for left < series <= right).

        Returns
        -------
        is_between : Series
        """
        if inclusive:
            try:
                pair = iter(inclusive)
                left_inclusive = pair.next()
                rigt_inclusive = pair.next()
                lmask = self >= left if left_inclusive else self > left
                rmask = self <= right if rigt_inclusive else self < right
            except TypeError:
                lmask = self >= left
                rmask = self <= right
        else:
            lmask = self > left
            rmask = self < right

        return lmask & rmask

```

**Comment by jorisvandenbossche** [comment]
@jreback yes, this method already exists ... (didn't know that! so my first reaction was the same as yours) 

So given that, I don't see any harm is extending its functionality a little bit

**Comment by jreback** [comment]
I suppose ok, but let's deprecate this method entirely. This is just another (confusing) way of doing things. There are already too many ways of doing indexing. This (although a nice name) is just bloating the API.

Pls create another issue for this deprecation (which will have to wait for 0.19.0)

**Comment by shoyer** [comment]
+1 for deprecating between

**Comment by jreback** [comment]
closed in favor of #13027
